### PR TITLE
Feature: Added "Edit in notepad" action to the right click context menu

### LINKED ISSUE #15412: Feature: Display "Edit" menu item for Batch Files in the main context menu
### What feature or improvement do you think would benefit Files?

The shell extension "Edit" is currently hidden away in the sub menu when right clicking batch files. We can improve the user experience by displaying this option in the main menu.

### Requirements

- Display "Edit" menu item in the main when right clicking batch files
- Position the menu item after "open"

### Files Version

v3.4.1

### Windows Version

Windows 11

### Comments

_No response_

### PR DESCRIPTION
**Pull request by yaira2**
**Resolved / Related Issues**

To prevent extra work, all changes to the Files codebase must link to an approved issue marked as `Ready to build`. Please insert the issue number following the hashtag with the issue number that this Pull Request resolves.
- Closes #15412

**Steps used to test these changes**

Stability is a top priority for Files and all changes are required to go through testing before being merged into the repo. Please include a list of steps that you used to test this PR.

1. Right clicked batch file and confirmed the "Edit in notepad" option is displayed on the main page

**Notes**
The sub menu still displays the "Edit in notepad" shell extension. We could filter it out but that'll require some more changes and it's out of scope for this PR.

### CODE REVIEW

**Review by hishitetsu — APPROVED** [review]
LGTM about this PR, but shouldn't "Edit in notepad" be available for CMD files and AHK files?

**Comment by yaira2** [comment]
> LGTM about this PR, but shouldn't "Edit in notepad" be available for CMD files and AHK files?

We can support those as well, it's essentially the same as the shell extension but easier to access.

File: src/Files.App/Actions/Open/EditInNotepadAction.cs [hunk 1]
```diff
@@ -20,7 +20,9 @@ public RichGlyph Glyph
 
 		public bool IsExecutable =>
 			context.SelectedItem is not null &&
-			FileExtensionHelpers.IsBatchFile(context.SelectedItem.FileExtension);
+			(FileExtensionHelpers.IsBatchFile(context.SelectedItem.FileExtension) ||
+			FileExtensionHelpers.IsAhkFile(context.SelectedItem.FileExtension) ||
+			FileExtensionHelpers.IsTextFile(context.SelectedItem.FileExtension));
```
Comment (hishitetsu) [inline src/Files.App/Actions/Open/EditInNotepadAction.cs:25]:
```suggestion
			FileExtensionHelpers.IsCmdFiles(context.SelectedItem.FileExtension));
```
I think text files doesn't need this option as they are opened in notepad by default.

Comment (yaira2) [reply src/Files.App/Actions/Open/EditInNotepadAction.cs]: I just came to the same realization!

**Review by hishitetsu — APPROVED** [review]

**Comment by yaira2** [comment]
Two more things I noticed
1. This action needs to be disabled in Recycle Bin
2. This action needs to be updated to support multiple files

**Comment by hishitetsu** [comment]
True.

**Comment by yaira2** [comment]
> This action needs to be updated to support multiple files

In theory this should work now but in practice it's not...

File: src/Files.App/Actions/FileSystem/CreateFolderWithSelectionAction.cs [hunk 2]
```diff
@@ -18,6 +18,8 @@ public RichGlyph Glyph
 
 		public bool IsExecutable =>
 			context.ShellPage is not null &&
+			context.PageType != ContentPageTypes.RecycleBin &&
+			context.PageType != ContentPageTypes.ZipFolder &&
```
Comment (yaira2) [inline src/Files.App/Actions/FileSystem/CreateFolderWithSelectionAction.cs:22]: There are probably other actions that need to be disabled in Recycle Bin.

File: src/Files.App/Actions/Open/EditInNotepadAction.cs [hunk 3]
```diff
@@ -20,7 +20,9 @@ public RichGlyph Glyph
 
 		public bool IsExecutable =>
 			context.SelectedItem is not null &&
```
Comment (hishitetsu) [inline src/Files.App/Actions/Open/EditInNotepadAction.cs:22]:
```suggestion
			context.SelectedItems.Any() &&
```
This should fix the issue with multiple selection.

File: src/Files.App/Actions/FileSystem/CreateFolderWithSelectionAction.cs (hunk 2 above)
Comment (hishitetsu) [reply src/Files.App/Actions/FileSystem/CreateFolderWithSelectionAction.cs:22]: We should check it, but it's a separate issue.

**Review by 0x5bfa — APPROVED** [review]
LGTM, codebase quality wise.

File: src/Files.App/Actions/FileSystem/CreateFolderWithSelectionAction.cs (hunk 2 above)
Comment (yaira2) [reply src/Files.App/Actions/FileSystem/CreateFolderWithSelectionAction.cs:22]: True

**Pull request merged**